File: src/LeetCode/Algorithms/Java/Jewels_and_Stones.java
package LeetCode.Algorithms.Java;

// https://leetcode.com/problems/jewels-and-stones/
public class Jewels_and_Stones {
	public static void main(String[] args) {
		String J = "z"; // "aA";
		String S = "ZZ"; // "aAAbbbb";

		System.out.println(numJewelsInStones(J, S));
	}

	public static int numJewelsInStones(String J, String S) {
		int result = 0;

		for (int i = 0; i < J.length(); i++) {
			for (int j = 0; j < S.length(); j++) {
				if (J.charAt(i) == S.charAt(j))
					result++;
			}
		}

		return result;
	}
}

File: src/LeetCode/Algorithms/JavaScript/Jewels_and_Stones.js
/**
 * https://leetcode.com/problems/jewels-and-stones/
 * @param {string} J
 * @param {string} S
 * @return {number}
 */
var numJewelsInStones = function(J, S) {
	return S.split('').reduce((a, c) => J.includes(c) ? a + 1 : a, 0);
};

let J = 'aA';
let S = 'aAAbbbb';

console.log(numJewelsInStones(J, S));